rfcs/0000-remove-builders.md

```diff
@@ -0,0 +1,34 @@
+Today, kpack has a complicated hierarchy of builder resources (Builder & ClusterBuilder & CustomBuilder & CustomClusterBuilder). Navigating this hierarchy is hard to explain and has consistently been confusing in early feedback. This is complicated by the recommended kpack experience being "CustomBuilders" when it appears that the obvious default is "Builder" or "ClusterBuilder".
```

Comment:
I agree that the overhead and confusion of having two resources is a significant concern.